services/arrow-flight/src/main/java/com/dremio/service/flight/DremioFlightProducer.java

```diff
@@ -333,14 +333,23 @@ public void getStreamCatalogs(CallContext callContext, Ticket ticket,
   public FlightInfo getFlightInfoSchemas(CommandGetSchemas commandGetSchemas,
                                          CallContext callContext,
                                          FlightDescriptor flightDescriptor) {
-    throw CallStatus.UNIMPLEMENTED.withDescription("CommandGetSchemas not supported.").toRuntimeException();
+    final Schema schema = getSchemaSchemas().getSchema();
+    return getFlightInfoForFlightSqlCommands(commandGetSchemas, flightDescriptor, schema);
   }
 
   @Override
   public void getStreamSchemas(CommandGetSchemas commandGetSchemas,
                                CallContext callContext, Ticket ticket,
                                ServerStreamListener serverStreamListener) {
-    throw CallStatus.UNIMPLEMENTED.withDescription("CommandGetSchemas not supported.").toRuntimeException();
+    final CallHeaders headers = retrieveHeadersFromCallContext(callContext);
```

Comment:
Do you need the headers anywhere but the subsequent call to get the UserSession? Can you collapse into one call?

Reply:
Fixed